src/ssl.c

```diff
@@ -208,13 +208,15 @@
 #define WOLFSSL_EVP_INCLUDED
 #include "wolfcrypt/src/evp.c"
 
+/* Crypto code uses EVP APIs. */
+#define WOLFSSL_SSL_CRYPTO_INCLUDED
+#include "src/ssl_crypto.c"
```

Comment:
All of the refactoring looks good and test cases. I'd like to bring the naming of this file "ssl_crypto.c" and it's location up internally though before merging this in.